config/src/main/java/com/typesafe/config/impl/ConfigBeanImpl.java

```diff
@@ -116,8 +116,15 @@
                     if (isOptionalProperty(clazz, beanProp)) {
                         continue;
                     }
-                    // Otherwise, raise a {@link Missing} exception right here
-                    throw new ConfigException.Missing(beanProp.getName());
+                    // Otherwise, raise a {@link BugOrBroken} exception
+                    throw new ConfigException.BugOrBroken("Should have detected missing props earlier");
```

Comment:
This should be reverted to the original exception, because the related change (which was proposed [here](https://github.com/typesafehub/config/pull/441#discussion_r106462597)) is not in the PR anymore.

Reply:
Just saw this comment. Don't know why I didn't get an email notification earlier. In any case, I've reverted to the original exception. Sorry for the belated response.